WordPress/Classes/Models/Notifications/NotificationSettings.swift

```diff
@@ -0,0 +1,185 @@
+        */
+        public init(kind: String, preferences: [String : Bool]?) {
+            self.kind           = Kind(rawValue: kind) ?? .Email
+            self.preferences    = preferences
+        }
+        
+        
+        /**
+        *  @enum    Stream.Kind
+        *  @brief   Enumerates all of the possible Stream Kinds
+        */
+        public enum Kind : String {
+            case Timeline       = "timeline"
+            case Email          = "email"
+            case Device         = "device"
+            
+            
+            /**
+            *  @details Returns the localized description of the current enum value
+            */
+            func description() -> String {
+                switch self {
+                case .Timeline:
+                    return NSLocalizedString("Notifications Tab", comment: "WordPress.com Notifications Timeline")
+                case .Email:
+                    return NSLocalizedString("Email", comment: "Email Notifications Channel")
+                case .Device:
+                    return NSLocalizedString("Push Notifications", comment: "Mobile Push Notifications")
+                }
+            }
+            
+            static let allValues = [ Timeline, Email, Device ]
+        }
+    }
+    
+    // MARK: - Private Properties
+    private let blogPreferenceKeys      = ["new_comment", "comment_like", "post_like", "follow", "achievement", "mentions"]
+    private let blogEmailPreferenceKeys = ["new_comment", "comment_like", "post_like", "follow", "mentions"]
+    private let otherPreferenceKeys     = ["comment_like", "comment_reply"]
+    private let wpcomPreferenceKeys     = ["marketing", "research", "community"]
+    
+    // MARK: - Localized Setting Descriptions
+    private let localizedDescriptionMap = [
+        "new_comment"       : NSLocalizedString("Comments on my site",      comment: "Notifications Settings"),
+        "comment_like"      : NSLocalizedString("Likes on my comments",     comment: "Notifications Settings"),
+        "post_like"         : NSLocalizedString("Likes on my posts",        comment: "Notifications Settings"),
+        "follow"            : NSLocalizedString("Site follows",             comment: "Notifications Settings"),
+        "achievement"       : NSLocalizedString("Site achievements",        comment: "Notifications Settings"),
+        "mentions"          : NSLocalizedString("Username Mentions",        comment: "Notifications Settings"),
+        "comment_reply"     : NSLocalizedString("Replies to your comments", comment: "Notifications Settings"),
+        "marketing"         : NSLocalizedString("Suggestions",              comment: "Notifications Settings"),
+        "research"          : NSLocalizedString("Research",                 comment: "Notifications Settings"),
+        "community"         : NSLocalizedString("Community",                comment: "Notifications Settings")
```

Comment:
The comment for localized strings should probably uniquely identify the string. At least that's how we've used that before and it seems to work better with GlotPress.